src/Views/SilenceForm/Updates.elm

```diff
@@ -0,0 +1,150 @@
+
+import Views.SilenceForm.Types exposing (SilenceFormMsg(..))
+import Types exposing (Model, Msg(MsgForSilenceForm, NewUrl, PreviewSilence))
+import Utils.Types exposing (ApiResponse(Success, Loading, Failure))
+import Utils.List
+import Utils.Date
+import Silences.Api
+import Silences.Types exposing (nullMatcher, nullSilence)
+import Task
+import Time
+
+
+update : SilenceFormMsg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        AddMatcher silence ->
+            -- TODO: If a user adds two blank matchers and attempts to update
+            -- one, both are updated because they are identical. Maybe add a
+            -- unique identifier on creation so this doesn't happen.
+            ( { model | silence = Success { silence | matchers = silence.matchers ++ [ nullMatcher ] } }, Cmd.none )
+
+        CreateSilence silence ->
+            ( { model | silence = Loading }, Silences.Api.create silence (SilenceCreate >> MsgForSilenceForm) )
+
+        UpdateStartsAt silence time ->
+            -- TODO:
+            -- Update silence to hold datetime as string, on each pass through
+            -- here update an error message "this is invalid", but let them put
+            -- it in anyway.
+            let
+                startsAt =
+                    Utils.Date.timeFromString time
+
+                duration =
+                    Maybe.map2 (-) silence.endsAt.t startsAt.t
+                        |> Maybe.map Utils.Date.duration
+                        |> Maybe.withDefault silence.duration
+            in
+                ( { model | silence = Success { silence | startsAt = startsAt, duration = duration } }, Cmd.none )
+
+        UpdateEndsAt silence time ->
+            let
+                endsAt =
+                    Utils.Date.timeFromString time
+
+                duration =
+                    Maybe.map2 (-) endsAt.t silence.startsAt.t
+                        |> Maybe.map Utils.Date.duration
+                        |> Maybe.withDefault silence.duration
+            in
+                ( { model | silence = Success { silence | endsAt = endsAt, duration = duration } }, Cmd.none )
+
+        UpdateDuration silence time ->
+            let
+                duration =
+                    Utils.Date.durationFromString time
+
+                endsAt =
+                    Maybe.map2 (+) silence.startsAt.t duration.d
+                        |> Maybe.map Utils.Date.fromTime
+                        |> Maybe.withDefault silence.endsAt
+            in
+                ( { model | silence = Success { silence | duration = duration, endsAt = endsAt } }, Cmd.none )
+
+        UpdateCreatedBy silence by ->
+            ( { model | silence = Success { silence | createdBy = by } }, Cmd.none )
+
+        SilenceCreate silence ->
+            case silence of
+                Success id ->
+                    ( { model | silence = Loading }, Task.perform identity (Task.succeed <| NewUrl ("/#/silence/" ++ id)) )
```

Comment:
Yes for sure. @w0rm Is there a better way of just dispatching a new message in an update function?